internal/mailer/validate.go

```diff
@@ -0,0 +1,120 @@
+package mailer
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/mail"
+	"strings"
+	"time"
+)
+
+var invalidEmailMap = map[string]bool{
+	"test@gmail.com": true,
+	"test@test.com":  true,
+	"test@email.com": true,
+}
+
+// https://www.rfc-editor.org/rfc/rfc2606
+var invalidHostMap = map[string]bool{
+	"test":        true,
+	"example":     true,
+	"invalid":     true,
+	"local":       true,
+	"localhost":   true,
+	"example.com": true,
+	"example.net": true,
+	"example.org": true,
+}
+
+const (
+	validateEmailTimeout = 250 * time.Millisecond
```

Comment:
How was the final 250ms? No objection to it - thought it'd be good to know in case there's ever a need to change the value

If it's an empirical value that sounds sane to me as well.

Reply:
DNS is usually pretty fast, the important part is to have some reasonable upper bounds to protect against DNS honey pot attacks or just bad / slow resolvers.